README.md

```diff
@@ -129,45 +130,59 @@ client.request({
 })
 ```
 
-Abortion is supported by destroying the request or
-response body.
+Non-idempotent requests will not be pipelined in order
+to avoid indirect failures.
 
-```js
-// Abort while sending request.
-const body = new stream.Passthrough()
-const promise = client.request({
-  path: '/',
-  method: 'POST',
-  body
-})
-body.destroy()
-const { statusCode, headers } = await promise
-```
+Idempotent requests will be automatically retried if
+they fail due to indirect failure from the request
+at the head of the pipeline. This does not apply to
+idempotent requests with a stream request body.
+
+##### Aborting a request
+
+A request can may be aborted using either an `AbortController` or an `EventEmitter`.
+To use `AbortController`, you will need to `npm i abort-controller`.
 
 ```js
-// Abort while reading response.
-const { statusCode, headers, body } = await client.request({
+const { AbortController } = require('abort-controller')
+const { Client } = require('undici')
+
+const client = new Client'http://localhost:3000')
+const abortController = new AbortController()
+
+client.request({
   path: '/',
-  method: 'GET'
+  method: 'GET',
+  signal: abortController.signal
+}, function (err, data) {
+  console.log(err) // RequestAbortedError
+  client.close()
 })
-body.destroy()
+
+abortController.abort()
 ```
 
-Promises and async await are supported as well!
+Alternatively, any `EventEmitter` that emits an `'abort'` event may be used as an abort controller:
+
 ```js
-const { statusCode, headers, body } = await client.request({
+const EventEmitter = require('events')
+const { Client } = require('undici')
+
+
+const client = new Client'http://localhost:3000')
+const ee = new EventEmitter()
+
+client.request({
   path: '/',
-  method: 'GET'
+  method: 'GET',
+  signal: ee
+}, function (err, data) {
+  console.log(err) // RequestAbortedError
+  client.close()
 })
-```
-
-Non-idempotent requests will not be pipelined in order
-to avoid indirect failures.
 
-Idempotent requests will be automatically retried if
-they fail due to indirect failure from the request
-at the head of the pipeline. This does not apply to
-idempotent requests with a stream request body.
+ee.emit('abort')
+```
```

Comment:
I think you should mention that aborting the request or response body will also have the same effect.